src/test/java/uk/gov/companieshouse/api/accounts/controller/ApprovalControllerTest.java

```diff
@@ -0,0 +1,127 @@
+package uk.gov.companieshouse.api.accounts.controller;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.servlet.HandlerMapping;
+import uk.gov.companieshouse.api.accounts.AttributeName;
+import uk.gov.companieshouse.api.accounts.exception.DataException;
+import uk.gov.companieshouse.api.accounts.model.entity.CompanyAccountEntity;
+import uk.gov.companieshouse.api.accounts.model.rest.Approval;
+import uk.gov.companieshouse.api.accounts.model.rest.SmallFull;
+import uk.gov.companieshouse.api.accounts.service.impl.ApprovalService;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseObject;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseStatus;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
+import uk.gov.companieshouse.api.accounts.utility.ApiResponseMapper;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@TestInstance(Lifecycle.PER_CLASS)
+public class ApprovalControllerTest {
+
+    @Mock
+    private HttpServletRequest request;
+
+    @Mock
+    private Transaction transaction;
+
+    @Mock
+    private SmallFull smallFull;
+
+    @Mock
+    private CompanyAccountEntity companyAccountEntity;
+
+    @Mock
+    private Approval approval;
+
+    @Mock
+    private BindingResult bindingResult;
+
+    @Mock
+    private ApprovalService approvalService;
+
+    @Mock
+    private Map<String, String> links;
+
+    @Mock
+    private ApiResponseMapper apiResponseMapper;
+
+    @InjectMocks
+    private ApprovalController approvalController;
+
+    @BeforeEach
+    public void setUp() throws NoSuchAlgorithmException, DataException {
+        when(request.getAttribute("transaction")).thenReturn(transaction);
```

Comment:
Would it be possible to tidy this a bit somehow? Sorting by mock behaviour? 
Just wondering, do both tests need all these return behaviours?